src/openforms/forms/api/viewsets.py

```diff
@@ -590,7 +590,12 @@ def logic_rules_bulk_update(self, request, *args, **kwargs):
     @logic_rules_bulk_update.mapping.get
     def logic_rules_list(self, request, *args, **kwargs):
         form = self.get_object()
-        logic_rules = form.formlogic_set.all()
+
+        # Had to apply prefetch here as too many queries (n+1) were taking place
+        # See github issue https://github.com/open-formulieren/open-forms/issues/5077
```

Comment:
Can you put this in the commit message rather than as a comment? This is a clear "why" of the change :)